components/do_verify_email/do_verify_email.jsx

```diff
@@ -25,30 +31,30 @@ export default class DoVerifyEmail extends React.Component {
     }
 
     componentWillMount() {
-        verifyEmail(
-            this.props.location.query.token,
-            () => {
-                browserHistory.push('/login?extra=verified&email=' + encodeURIComponent(this.props.location.query.email));
-            },
-            (err) => {
-                let serverError;
-                if (err) {
-                    serverError = (
-                        <FormattedMessage
-                            id='signup_user_completed.invalid_invite'
-                            defaultMessage='The invite link was invalid.  Please speak with your Administrator to receive an invitation.'
-                        />
-                    );
-                }
-
-                this.setState({
-                    verifyStatus: 'failure',
-                    serverError
-                });
-            }
-        );
+        this.verifyEmail();
     }
 
+    verifyEmail = async () => {
+        const {actions: {verifyUserEmail}} = this.props;
+        const {data, error} = await verifyUserEmail(this.props.location.query.token);
+        let serverError;
```

Comment:
Super minor, but it looks like `serverError` is only used in the `else if`, mind moving the declaration inside there?